website/scripts/ros/ros_traffic_signal.js

```diff
@@ -1,67 +1,102 @@
+                                                        $('.traffic-signal-col').html(updateTrafficSignal('green',remaining_time));
+                                                        break;
+                                                    case TRAFFIC_SIGNAL_PHASE_STATE.stop_and_remain:
+                                                        $('.traffic-signal-col').html(updateTrafficSignal('red',remaining_time));
+                                                        break;
+                                                    case TRAFFIC_SIGNAL_PHASE_STATE.protected_clearance:
+                                                        $('.traffic-signal-col').html(updateTrafficSignal('yellow',remaining_time));
+                                                        break;
+                                                    case TRAFFIC_SIGNAL_PHASE_STATE.permissive_movement_allowed:
+                                                        $('.traffic-signal-col').html(updateTrafficSignal('flash_green',remaining_time));
+                                                        break;
+                                                    case TRAFFIC_SIGNAL_PHASE_STATE.permissive_clearance:
+                                                    case TRAFFIC_SIGNAL_PHASE_STATE.caution_conflicting_traffic:
+                                                        $('.traffic-signal-col').html(updateTrafficSignal('flash_yellow',remaining_time));
+                                                        break;
+                                                    case TRAFFIC_SIGNAL_PHASE_STATE.stop_then_proceed:
+                                                        $('.traffic-signal-col').html(updateTrafficSignal('flash_red',remaining_time));
+                                                        break;
+                                                    default:
+                                                        $('.traffic-signal-col').html(updateTrafficSignal('',''));
+                                                        console.error("Traffic signal state is invalid");
+                                                        break;
+                                                } 
+                                        }
+                                    });
+                                }
+                            
+                        });
+                    }                    
+                });
+            }
+         }catch(error){
+            console.error(error);
+         }  
+
+         //set back to black after 1 second.
+        if(!isTimeerSet)
+        {
+            isTimeerSet=true;
+            setTimeout(function(){
+                $('.traffic-signal-col').html(updateTrafficSignal('',''));
+                isTimeerSet=false;
+                signalState = null;
+            }, 1000);
```

Comment:
I'm still not clear on how this timer won't cause the light to expire after 1 second. Doesn't there need to be some last message time to compare against?